hack/verify-golangci-lint-pr.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script checks a PR for the coding style for the Go language files using
+# golangci-lint. It does nothing when invoked as part of a normal "make
+# verify".
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [ ! "${PULL_NUMBER}" ]; then
+    echo 'Not testing anything because this is not a pull request.'
+    exit 0
+fi
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+exec "${KUBE_ROOT}/hack/verify-golangci-lint.sh" -r "${PULL_BASE_SHA}" -s -g "${ARTIFACTS}/golangci-lint-githubactions.log"
```

Comment:
I thought githubactions was not working

Reply:
This PR is work-in-progress. Integration with GitHub is tracked in https://github.com/kubernetes/test-infra/issues/17056. I would love to see that resolved before enable stricter checking because otherwise contributors will have to read the text output of the Prow job to find out where the linter is complaining.

But if that those enhancements in Prow are taking too long, perhaps we need a plan B - not sure yet.

Reply:
Let's move forward with a separate, optional, text-only "pull-kubernetes-verify-strict-lint" job.

This PR has been updated accordingly (just need to rebase once to get rid of commits from the "go vet" PR) and I have created a PR for that job in https://github.com/kubernetes/test-infra/pull/28914.

Reply:
Both PRs are ready for review and approval.